memestra/memestra.py

```diff
@@ -189,9 +202,13 @@ def filter_out(user):
                                       if isinstance(n, _defs)]
                     if any(f in self.deprecated for f in user_ancestors):
                         continue
+
+                    user_ancestor = user_ancestors[-1] if user_ancestors else user.node
```

Comment:
Maybe we should comment here on what's `user_ancestors[-1]` is. Something like... "Starts on the deepest function". -or, I don't really know how to phrase it better...
But I think it'd make the code clearer, cause it's not so obvious what's happening here, imo.